### ISSUE TITLE
[BUG] zome compiled to wasm can't be installed in holochain, due to dependency issue

### ISSUE THREAD

**Issue by Connoropolous** [holochain/holochain#1512]
**Describe the bug**
@ddd-mtl initially posted this:
"In HDK 138, when I use #[hdk_entry_defs] or #[hdk_link_types] in the integrity zome, I get the following error when generating a sandbox:"
Myself, and @zippy also experienced this.
The exact error is:
```
Error: Failed to install app: Expected AdminResponse::AppBundleInstalled but got Error(InternalError("Conductor returned an error while using a ConductorApi: RibosomeError(WasmRuntimeError(RuntimeError { source: User(WasmError { file: \"crates/holochain/src/core/ribosome/real_ribosome.rs\", line: 416, error: Compile(\"Error while importing \\\"__wbindgen_placeholder__\\\".\\\"__wbindgen_describe\\\": unknown import. Expected Function(FunctionType { params: [I32], results: [] })\") }), wasm_trace: [], native_trace:    0: <unknown>\n   1: <unknown>\n   2: <unknown>\n   3: <unknown>\n   4: <unknown>\n   5: <unknown>\n   6: <unknown>\n   7: <unknown>\n   8: <unknown>\n   9: <unknown>\n  10: <unknown>\n  11: <unknown>\n  12: <unknown>\n  13: <unknown>\n  14: <unknown>\n  15: <unknown>\n  16: <unknown>\n  17: <unknown>\n  18: <unknown>\n  19: <unknown>\n  20: <unknown>\n  21: <unknown>\n  22: <unknown>\n  23: <unknown>\n  24: <unknown>\n  25: <unknown>\n  26: <unknown>\n  27: <unknown>\n  28: <unknown>\n  29: <unknown>\n  30: <unknown>\n }))"))
```
Key message: `Error while importing \\\"__wbindgen_placeholder__\\\".\\\"__wbindgen_describe\\\": unknown import. Expected Function(FunctionType { params: [I32], results: [] }`

**Apparent Cause**
"Found my problem by comparing with elemental-chat's cargo.lock"
"cargo picks up latest minor version of chrono  which added a dependency to wasm-bindgen"
This messes with the wasmer guest/host interface between holochain and the wasm

**Workaround**
(this can be useful for app devs in the meantime)
So the actual workaround is to add  this chrono dependency to my cargo.toml: chrono = "=0.4.6"

**Potential Solution For Holochain**
An actual fix in holochain would be to change kitsune_p2p_timestamp's dependency to chrono from chrono = "0.4.6" to chrono = "=0.4.6"

**Expected behavior**
For it to install the bundle successfully

**System information:**
HDK 138

**Comment by abe-njama** [comment]
I've taken note of this issue. Flagging it to the attention of the internal team.

**Comment by Connoropolous** [comment]
my take... 

instead of doing that, it would probably be ideal to get an accurate read on the chrono API including the Rust "features" it has, and configuring it (if possible) to not have the wasm-bindgen dependency. 

What I've snooped so far is that `chrono` changed its Rust features, and made the 'wasm-bidngen' dependency mandatory and default, instead of optional, and enabled with a feature flag. So that sucks. Maybe they can be appealed to adjust that change and still make it 'optional' but enabled by default, with a feature to 'disable'

The '= 0.4.6' should be a shorter term 'workaround', as we are already doing 'downstream' in our apps

**Comment by neonphog** [comment]
chrono has off and on been a security vulnerability... maybe at this point it'd be better to try to write it out of our code as a dependency

**Comment by mhuesch** [comment]
I am curious if anyone has a detailed understanding of what causes this problem :thinking: 

I am running into it on my secret quest to build a Rust/wasm frontend for a hApp.

I tried to factor out some code which is common between my frontend and backend (hApp / zome code), into a new crate, but once I added `wasm-bindgen` to said crate's deps, I encountered this error.

it is quite awkward b/c it means I cannot librarify any functions which have `wasm-bindgen` library types in them...

wondering if there could be a way to work around the problem, or if hApps having secondary dependencies on `wasm-bindgen` is a total no-go.

any help appreciated (also if anyone knows how to view the generated wasm and see what the actual problem is at the wasm level) :pray:

**Comment by maackle** [comment]
Just merged @freesig's #1527, hopefully this is fixed now

**Comment by jost-s** [comment]
> I am curious if anyone has a detailed understanding of what causes this problem 🤔

Now running into this bug while working on Tryorama, I'm wondering the same. Pinning to chrono `0.4.6` is no longer possible, as it has a lower version serde as a dependency. The fix is to disable default features in `chrono`, which pulls in wasm-bindgen.

I don't know why that causes a problem. Holochain uses wasm-bindgen as well, but even pinning to the same version in Tryorama throws up this error at the zome call threshold
```
Error while importing \"__wbindgen_placeholder__\".\"__wbindgen_describe\": unknown import. Expected Function(FunctionType { params: [I32]
```
